sale_lead_time/models/sale_order.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2025 Quartile
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    delivery_lead_time = fields.Float(
+        compute="_compute_delivery_lead_time",
+        store=True,
+        readonly=False,
+        help="Number of days expected for delivery from the warehouse to the delivery address.",
+    )
+
+    def _get_partner_address_domain(self):
+        self.ensure_one()
+        partner = self.partner_shipping_id
+        partner_domain = [("partner_id", "=", partner.id)]
+        state_domain = [
+            ("partner_id", "=", False),
+            ("state_id", "=", partner.state_id.id),
+        ]
+        country_domain = [
+            ("partner_id", "=", False),
+            ("state_id", "=", False),
+            ("country_id", "=", partner.country_id.id),
+        ]
+        no_address_domain = [
+            ("partner_id", "=", False),
+            ("state_id", "=", False),
+            ("country_id", "=", False),
+        ]
+        return expression.OR(
+            [partner_domain, state_domain, country_domain, no_address_domain]
+        )
+
+    def _get_warehouse_domain(self):
+        self.ensure_one()
+        return [
+            "|",
+            ("warehouse_id", "=", self.warehouse_id.id),
+            ("warehouse_id", "=", False),
+        ]
+
+    @api.depends("partner_shipping_id", "warehouse_id")
+    def _compute_delivery_lead_time(self):
+        for rec in self:
+            rec.delivery_lead_time = 0.0
+            if not rec.partner_shipping_id:
+                continue
+            partner_address_domain = rec._get_partner_address_domain()
+            warehouse_domain = rec._get_warehouse_domain()
+            domain = expression.AND([partner_address_domain, warehouse_domain])
```

Comment:
Forgot to include the company in the domain.
```suggestion
            domain = expression.AND([partner_address_domain, warehouse_domain, [("company_id", "=", rec.company_id.id)]])
```